### PR TITLE
OHRI-1857 : Users should only be able to edit the latest MNCH visit/e…

### PR DESCRIPTION
**Pull request by Herobiam**
…ncounter

## Requirements

- [ ] This PR has a title that briefly describes the work done including the ticket number. If there is a ticket, make sure your PR title includes the ticket number in the format `OHRI-123 My PR title`.
- [ ] My work includes tests or is validated by existing tests.

## Summary
[OHRI-1857](https://ohri.atlassian.net/jira/software/projects/OHRI/boards/1?assignee=640074db9cba7ca02877c179&selectedIssue=OHRI-1857)

This feature should not allow users to go back and edit fields from a previous encounters. E.g if a pregnancy has a first ANC entered and a follow-up ANC encounter/visit added then user should not be able to edit the 1st ANC but editing for the latest follow up encounter/visit should still be possible.

## Screenshots
https://github.com/UCSF-IGHS/openmrs-esm-ohri/assets/19629227/405b516c-c6f0-4d7e-8e9c-f68367fc8619

## Other

[OHRI-1857]: https://ohri.atlassian.net/browse/OHRI-1857?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: packages/esm-commons-lib/src/components/encounter-list/encounter-list.component.tsx [hunk 1]
```diff
@@ -124,16 +124,17 @@ export const EncounterList: React.FC<EncounterListProps> = ({
 
   const constructPaginatedTableRows = useCallback(
     (encounters: OpenmrsEncounter[], currentPage: number, pageSize: number) => {
-      const startIndex = (currentPage - 1) * pageSize;
+      console.log('counters', encounters);
```
Comment (ODORA0) [inline packages/esm-commons-lib/src/components/encounter-list/encounter-list.component.tsx:127]: @Herobiam You need to remove this log

**Review by ODORA0 — CHANGES_REQUESTED** [review]

File: packages/esm-commons-lib/src/components/encounter-list/encounter-list.component.tsx (hunk 1 above)
Comment (Herobiam) [reply packages/esm-commons-lib/src/components/encounter-list/encounter-list.component.tsx]: Thanks @ODORA0

**Comment by ODORA0** [comment]
@Herobiam The requirement of the issue says this has to be default to MNCH but am pretty sure this will affect all encounter lists since these changes are in the encounter list component in the shared commons lib

**Comment by Herobiam** [comment]
@ODORA0 i have made changes as per your comment above

**Review by ODORA0 — DISMISSED** [review]
LGTM

File: packages/esm-commons-lib/src/components/encounter-list/encounter-list.component.tsx [hunk 2]
```diff
@@ -124,16 +126,16 @@ export const EncounterList: React.FC<EncounterListProps> = ({
 
   const constructPaginatedTableRows = useCallback(
     (encounters: OpenmrsEncounter[], currentPage: number, pageSize: number) => {
-      const startIndex = (currentPage - 1) * pageSize;
+            const startIndex = (currentPage - 1) * pageSize;
```
Comment (pirupius) [inline packages/esm-commons-lib/src/components/encounter-list/encounter-list.component.tsx:129]: Please fix this indentation

File: packages/esm-commons-lib/src/components/encounter-list/encounter-list.component.tsx [hunk 3]
```diff
@@ -67,7 +69,7 @@ export const EncounterList: React.FC<EncounterListProps> = ({
     onFormSave,
   } = useEncounterRows(patientUuid, encounterType, filter);
   const { moduleName, workspaceWindowSize, displayText, hideFormLauncher } = launchOptions;
-
+
```
Comment (pirupius) [inline packages/esm-commons-lib/src/components/encounter-list/encounter-list.component.tsx:72]: Make sure your editor lints files on save and if configured correctly to avoid unnecessary changes like this extra space

File: packages/esm-commons-lib/src/components/encounter-list/encounter-list.component.tsx [hunk 4]
```diff
@@ -180,7 +182,7 @@ export const EncounterList: React.FC<EncounterListProps> = ({
         });
         // If custom config is available, generate actions accordingly; otherwise, fallback to the default actions.
         const actions = tableRow.actions?.length ? tableRow.actions : defaultActions;
-        tableRow['actions'] = (
+                tableRow['actions'] = (
```
Comment (pirupius) [inline packages/esm-commons-lib/src/components/encounter-list/encounter-list.component.tsx:185]: Run the linter on this whole file to fix all occurrences

File: packages/esm-commons-lib/src/components/encounter-list/encounter-list.component.tsx [hunk 5]
```diff
@@ -198,6 +200,7 @@ export const EncounterList: React.FC<EncounterListProps> = ({
                     workspaceWindowSize,
                   );
                 }}
+                disabled={actionItem.label == 'Edit form' && disableEdit == true && findEncounterLatestDateIndex(encounters) != encounterIndex}
```
Comment (pirupius) [inline packages/esm-commons-lib/src/components/encounter-list/encounter-list.component.tsx:203]: This logic looks very specific and doesn't look like something that can be reused because

1. The condition to disable the action seems to be based off the words `Edit form` which would break the moment the text changes
2. This doesn't play well with scenarios where actions are from multiple forms

My suggestion would be moving this logic away from here and adding it to the actions with the individual programs.

**Review by pirupius — CHANGES_REQUESTED** [review]

**Comment by pirupius** [comment]
PR moved to https://github.com/UCSF-IGHS/openmrs-esm-ohri/pull/1772
